lib/vernier.rb

```diff
@@ -31,9 +31,44 @@ def self.profile(mode: :wall, out: nil, gc: true, **collector_options)
     result
   end
 
+  @collector = nil
+  @collector_out = nil
+
+  def self.start_profile(mode: :wall, out: nil, gc: true, **collector_options)
+    if @collector
+      @collector.stop
+      @collector = @collector_out = nil
+
+      raise "Profile already started, stopping..."
+    end
+
+    gc &&= (mode == :retained)
+    3.times { GC.start } if gc
+
+    @collector_out = out
+    @collector = Vernier::Collector.new(mode, collector_options)
+    @collector.start
+  end
+
+  def self.stop_profile
+    raise "No profile started" unless @collector
+
+    result = @collector.stop
+    if @collector_out
+      result.write(out: @collector_out)
+    end
+    @collector = @collector_out = nil
+
+    result
+  end
+
   class << self
     alias_method :trace, :profile
     alias_method :run, :profile
+    alias_method :start_trace, :start_profile
+    alias_method :start_run, :start_profile
+    alias_method :stop_trace, :stop_profile
+    alias_method :stop_run, :stop_profile
```

Comment:
I think we should drop these. `start_profile`/`stop_profile` is probably sufficient as an interface.

```suggestion
```

Reply:
Sounds good, I'll address when I get a chance.

Reply:
Removed in [these changes](https://github.com/jhawthorn/vernier/compare/19e68dd55b8a09ebe527666a20b503558829cebb..7b2ee9603b1cb9908d97819e55c9082dc90aaf3c).